logger/handler/ringbuffer.go

```diff
@@ -0,0 +1,149 @@
+package handler
+
+import (
+	"container/ring"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"sync"
+
+	"github.com/deis/deis/logger/drain"
+	"github.com/deis/deis/logger/syslog"
+)
+
+var logStorage = make(map[string]*ring.Ring)
+var ringBufferSize int
+
+var regexpForPost = regexp.MustCompile(`^/([-a-z0-9]+)/.*`)
+var regexpForGet = regexp.MustCompile(`^/([-a-z0-9]+)/([0-9]+)/.*`)
+
+var mu sync.Mutex
+
+// Add log message to main map with ring byffers by project name
+func addToStorage(name string, message string) {
+	mu.Lock()
+	defer mu.Unlock()
+	currentRing, ok := logStorage[name]
+	if !ok {
+		r := ring.New(ringBufferSize)
+		r.Value = message
+		logStorage[name] = r
+	} else {
```

Comment:
Add a return statement to the end of the `if !ok { }` block then drop the else statement in here.